Sources/Commands/SwiftPackageRegistryTool.swift

```diff
@@ -0,0 +1,185 @@
+import Workspace
+import Foundation
+import PackageRegistry
+
+private enum RegistryConfigurationError: Swift.Error {
+    case missingScope(String? = nil)
+    case invalidURL(String)
+}
+
+extension RegistryConfigurationError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .missingScope(let scope?):
+            return "no existing entry for scope: \(scope)"
+        case .missingScope:
+            return "no existing entry for default scope"
+        case .invalidURL(let url):
+            return "invalid URL: \(url)"
+        }
+    }
+}
+
+public struct SwiftPackageRegistryTool: ParsableCommand {
+    public static var configuration = CommandConfiguration(
+        commandName: "package-registry",
+        _superCommandName: "swift",
+        abstract: "Interact with package registry and manage related configuration",
+        discussion: "SEE ALSO: swift package",
+        version: SwiftVersion.currentVersion.completeDisplayString,
+        subcommands: [
+            Set.self,
+            Unset.self
+        ],
+        helpNames: [.short, .long, .customLong("help", withSingleDash: true)])
+
+    @OptionGroup()
+    var swiftOptions: SwiftToolOptions
+
+    public init() {}
+
+    struct Set: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Set a custom registry")
+
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+
+        @Flag(help: "Apply settings to all projects for this user")
+        var global: Bool = false
+
+        @Option(help: "Associate the registry with a given scope")
+        var scope: String?
+
+        // TODO: Uncomment once .netrc management is implemented
+
+        // @Option(help: "Specify a user name for the remote machine")
+        // var login: String?
+
+        // @Option(help: "Supply a password for the remote machine")
+        // var password: String?
+
+        @Argument(help: "The registry URL")
+        var url: String
+
+        func run(_ swiftTool: SwiftTool) throws {
+            guard let url = URL(string: self.url),
+                  url.scheme == "https"
+            else {
+                throw RegistryConfigurationError.invalidURL(self.url)
+            }
+
+            // TODO: Require login if password is specified
+
+            let path = try swiftTool.getRegistryConfigurationPath(fileSystem: localFileSystem, global: global)
+
+            var configuration = RegistryConfiguration()
+            if localFileSystem.exists(path) {
+                configuration = try RegistryConfiguration.readFromJSONFile(in: localFileSystem, at: path)
+            }
+
+            if let scope = scope {
+                configuration.scopedRegistries[scope] = .init(url: url)
+            } else {
+                configuration.defaultRegistry = .init(url: url)
+            }
+
+            try configuration.writeToJSONFile(in: localFileSystem, at: path)
```

Comment:
I dont think its super likely, but we could use a file lock to make sure no concurrent access to the configuration file

Reply:
`writeToJSONFile` does an atomic write, so I _think_ we should be alright? If not, do you have any examples of the access pattern you're looking for?